gem/ci/build.ps1

```diff
@@ -0,0 +1,180 @@
+$ErrorActionPreference = 'Stop'
+
+function Get-CurrentDirectory
+{
+    $thisName = $MyInvocation.MyCommand.Name
+    [IO.Path]::GetDirectoryName((Get-Content function:$thisName).File)
+}
+
+function Get-ContainerVersion
+{
+    # Don't need to fetch anything else when running in Azure CI, as that will
+    # already have pulled everything down, and more importantly git-fetch will
+    # block waiting for authorization. TF_BUILD is set by Azure CI.
+    if ($ENV:TF_BUILD -ine 'true')
+    {
+        # shallow repositories need to pull remaining code to `git describe` correctly
+        if (Test-Path "$(git rev-parse --git-dir)/shallow")
+        {
+            git pull --unshallow
+        }
+
+        # tags required for versioning
+        git fetch origin 'refs/tags/*:refs/tags/*'
+    }
+
+    (git describe) -replace '-.*', ''
+}
+
+function Lint-Dockerfile($Path)
+{
+    hadolint --ignore DL3008 --ignore DL3018 --ignore DL4000 --ignore DL4001 $Path
+}
+
+function Build-Container(
+    $Name,
+    $Namespace = 'puppet',
+    $Context = $Name,
+    $Version = (Get-ContainerVersion),
+    $Vcs_ref = $(git rev-parse HEAD))
+{
+    Push-Location (Join-Path (Get-CurrentDirectory) '..')
+
+    $build_date = (Get-Date).ToUniversalTime().ToString('o')
+    $docker_args = @(
+        '--pull',
```

Comment:
This will need to be parameterizable since we don't wan't to `--pull` for the puppetserver container (we want to use the puppetserver-standalone container we just built)

Reply:
Yep, makes sense.

Puppetserver repo will cause some changes in here since it has multiple images in it, and thus a different file system structure than our other repos.